### PR TITLE
[lldb] Add the lldb bit to the pointer authentication mask

### PR DESCRIPTION
**Pull request by augusto2112**
rdar://89162577

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp [hunk 1]
```diff
@@ -24,6 +24,9 @@ bool LLDBMemoryReader::queryDataLayout(DataLayoutQueryType type, void *inBuffer,
       return false;
     // The mask returned by the process masks out the non-addressable bits.
     uint64_t mask_pattern = ~ptrauth_mask;
+    // Keep the lldb file address bit, so we discern between real and 
+    // lldb's vertiual address space.
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp]: virtual

Comment (adrian-prantl) [reply lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp]: How about: LLDBMemoryReader

Comment (adrian-prantl) [reply lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp]: How about: `LLDBMemoryReader sets LLDB_FILE_ADDRESS_BIT to distinguish process addresses and file addresses that point into a reflection section on disk. Setting the bit in the mask ensures it isn't accidentally cleared by ptrauth stripping.`

**Review by adrian-prantl — APPROVED** [review]
I think this is great! I left a suggestion for a better wording.

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**